### PR TITLE
Remainder negative numerator bug fixed

### LINKED ISSUE #640: [BUG] Remainder does not work properly when the numerator is negative
**Describe the bug**
The remainder does not function properly when the numerator is negative.

**To Reproduce**

Include code snippet
```python
import mlx.core as mx
print("Result:", (-mx.ones(1))%5)
```
Prints:
"Result: array([-1], dtype=float32)"

```python
import numpy as np
print("Result:",(-np.ones(1))%5)
```
Prints:
"Result: [4.]"

**Expected behavior**
Same result with Numpy

 - OS Version MacOS 14.2.1
 - Version: 0.1.0

### PR DESCRIPTION
**Pull request by abdussamettrkr**
## Proposed changes
Fix  remainder negative numerator is negative bug fixes #640.

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: mlx/backend/metal/kernels/binary.metal [hunk 1]
```diff
@@ -15,10 +15,33 @@ struct Divide {
...
+      r+= y;
+    return r; 
+  }
+  template <> complex64_t operator()(complex64_t x, complex64_t y) {
+    return x % y; 
+  }
+  template <> float operator()(float x, float y) {
+    float r = fmod(x, y);
+    if (r < 0 && y >0)
+      r+= y;
```
Comment (angeloskath) [inline mlx/backend/metal/kernels/binary.metal:30]: Spaces after `r` and `y >`. Possibly we also need an unsigned numbers specialization that doesn't do the check. Otoh the compiler should be able to optimize that away so perhaps we are good.

**Review by angeloskath — CHANGES_REQUESTED** [review]
Thanks for fixing this. I left a couple of comments, almost exclusively stylistic.

Feel free to ignore the unsigned inputs specialization since the compiler can most probably handle that case.

File: mlx/backend/metal/kernels/binary.metal (hunk 1 above)
Comment (abdussamettrkr) [reply mlx/backend/metal/kernels/binary.metal]: I added specialization for eval_cpu and eval, but i am not sure how to do it in metal.

File: mlx/backend/accelerate/primitives.cpp [hunk 2]
```diff
@@ -298,13 +298,22 @@ struct RemainderFn {
   std::enable_if_t<!std::is_integral_v<T>, T> operator()(
       T numerator,
       T denominator) {
-    return std::fmod(numerator, denominator);
+    auto r = std::fmod(numerator, denominator);
```
[the remark this answers is not in the document]
Comment (abdussamettrkr) [reply mlx/backend/accelerate/primitives.cpp]: I suppose I did

**Comment by abdussamettrkr** [comment]
Closes also #650

**Review by angeloskath — APPROVED** [review]
I added the Metal specialization, an extra test, rebased on main and am merging. Thanks for fixing this!

**Pull request merged**